cpp/oneapi/dal/backend/primitives/reduction/reduction_dpc.hpp

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/reduction/functors.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+/// Reduces `input` rows inplace and stores results into `output`
```

Comment:
"inplace" typically means that it overwrites the input data, in this case `input` are not overwritten. For example, function `syevd` is in-place, the eigenvectors are written to the same memory as input matrix, see https://spec.oneapi.com/versions/latest/elements/oneMKL/source/domains/lapack/syevd.html

Reply:
Fixed

Reply:
It's not fixed. The same for other places where "inplace" used in doxygen

Reply:
Fixed:
```
/// Reduces `input` rows and stores results into `output`
///
```

Reply:
Fixed